src/release.rs

```diff
@@ -31,19 +31,68 @@ pub struct TripleRelease {
     pub install_only_suffix: &'static str,
     /// Minimum Python version this triple is released for.
     pub python_version_requirement: Option<VersionSpecifier>,
+    /// Additional build suffixes to release conditional on the Python version.
+    pub conditional_suffixes: Vec<ConditionalSuffixes>,
+}
+
+/// Describes additional build suffixes conditional on the Python version.
+///
+/// e.g., free-threaded builds which are only available for Python 3.13+.
+pub struct ConditionalSuffixes {
+    /// The minimum Python version to include these suffixes for.
+    pub python_version_requirement: VersionSpecifier,
+    /// Build suffixes to release.
+    pub suffixes: Vec<&'static str>,
+}
+
+impl TripleRelease {
+    pub fn suffixes<'a>(
+        &'a self,
+        python_version: Option<&'a pep440_rs::Version>,
+    ) -> impl Iterator<Item = &'static str> + 'a {
+        self.suffixes
+            .iter()
+            .copied()
+            .chain(
+                self.conditional_suffixes
+                    .iter()
+                    .flat_map(move |conditional| {
+                        if python_version.is_none()
+                            || python_version.is_some_and(|python_version| {
+                                conditional
+                                    .python_version_requirement
+                                    .contains(python_version)
+                            })
+                        {
+                            conditional.suffixes.iter().copied()
+                        } else {
+                            [].iter().copied()
+                        }
+                    }),
+            )
+    }
 }
 
 pub static RELEASE_TRIPLES: Lazy<BTreeMap<&'static str, TripleRelease>> = Lazy::new(|| {
     let mut h = BTreeMap::new();
 
     // macOS.
     let macos_suffixes = vec!["debug", "pgo", "pgo+lto"];
+    let macos_suffixes_313 = vec![
+        "freethreaded+debug",
+        "freethreaded+pgo",
+        "freethreaded+pgo+lto",
+    ];
```

Comment:
Needs to be static so we can't derive it from the above list without a more invasive change.